src/Model/Scope/CompoundCondition.php

```diff
@@ -0,0 +1,199 @@
+
+            if ($nestedCondition->isEmpty()) {
+                continue;
+            }
+
+            $this->add(clone $nestedCondition);
+        }
+    }
+
+    public function __clone()
+    {
+        foreach ($this->elements as $k => $nestedCondition) {
+            $this->elements[$k] = clone $nestedCondition;
+            $this->elements[$k]->owner = $this;
+        }
+    }
+
+    /**
+     * Return array of nested conditions.
+     *
+     * @return AbstractCondition[]
+     */
+    public function getNestedConditions()
+    {
+        return $this->elements;
+    }
+
+    public function onChangeModel(): void
+    {
+        foreach ($this->elements as $nestedCondition) {
+            $nestedCondition->onChangeModel();
+        }
+    }
+
+    public function isEmpty(): bool
+    {
+        return empty($this->elements);
+    }
+
+    public function isCompound(): bool
+    {
+        return count($this->elements) > 1;
+    }
+
+    /**
+     * @return self::AND|self::OR
+     */
+    public function getJunction(): string
+    {
+        return $this->junction;
+    }
+
+    /**
+     * Checks if junction is OR.
+     */
+    public function isOr(): bool
+    {
+        return $this->junction === self::OR;
+    }
+
+    /**
+     * Checks if junction is AND.
+     */
+    public function isAnd(): bool
+    {
+        return $this->junction === self::AND;
+    }
+
+    /**
+     * Clears the group from nested conditions.
+     *
+     * @return static
+     */
+    public function clear()
+    {
+        foreach ($this->elements as $nestedCondition) {
+            $nestedCondition->destroy();
+        }
+
+        return $this;
+    }
+
+    public function simplify()
+    {
+        if (count($this->elements) != 1) {
+            return $this;
+        }
+
+        /** @var AbstractCondition $component */
+        $component = reset($this->elements);
+
+        return $component->simplify();
+    }
+
+    /**
+     * Use De Morgan's laws to negate.
+     *
+     * @return static
+     */
+    public function negate()
+    {
+        $this->junction = $this->junction == self::OR ? self::AND : self::OR;
+
+        foreach ($this->elements as $nestedCondition) {
+            $nestedCondition->negate();
+        }
+
+        return $this;
+    }
+
+    public function toWords(bool $asHtml = false): string
+    {
+        $parts = [];
+        foreach ($this->elements as $nestedCondition) {
+            $words = $nestedCondition->toWords($asHtml);
+
+            $parts[] = $this->isCompound() && $nestedCondition->isCompound() ? "({$words})" : $words;
+        }
+
+        $glue = ' ' . strtolower($this->junction) . ' ';
+
+        return implode($glue, $parts);
+    }
+
+    /**
+     * Merge number of conditions using AND as junction.
+     *
+     * @param AbstractCondition $_
+     *
+     * @return static
+     */
+    public static function mergeAnd(AbstractCondition $conditionA, AbstractCondition $conditionB, $_ = null)
```

Comment:
why not ...$conditions ?

Reply:
now I see what you want, your signature was probably better if `AbstractCondition` type would be used also for the 3rd type

but why we need mergeAnd/Or at all?

for me, it implements nesting which is supported in Scopes by definition

Reply:
You can see in the method what it is a shortcut for.

Reply:
resolved by rename to createAnd/Or